rotkehlchen/chain/evm/decoding/velodrome/balances.py

```diff
@@ -34,6 +52,63 @@ def __init__(
             deposit_event_types={(HistoryEventType.DEPOSIT, HistoryEventSubType.DEPOSIT_FOR_WRAPPED)},  # noqa: E501
             gauge_deposit_event_types={(HistoryEventType.DEPOSIT, HistoryEventSubType.DEPOSIT_ASSET)},  # noqa: E501
         )
+        self.protocol_token = protocol_token
+        self.voting_escrow_address = voting_escrow_address
 
     def get_gauge_address(self, event: 'EvmEvent') -> ChecksumEvmAddress | None:
         return event.address
+
+    def query_balances(self) -> BalancesSheetType:
+        balances = super().query_balances()
+        if len(addresses_with_deposits := self.addresses_with_activity(event_types=self.deposit_event_types)) == 0:  # noqa: E501
+            return balances
+
+        addresses_to_token_ids = {}
+        for address, events in addresses_with_deposits.items():
+            addresses_to_token_ids[address] = [event.extra_data['token_id'] for event in events if event.extra_data is not None]  # noqa: E501
```

Comment:
```suggestion
        addresses_to_token_ids = {
            address: [event.extra_data['token_id'] for event in events if event.extra_data is not None]
            for address, events in addresses_with_deposits.items()
        }
```
Also, will `extra_data['token_id']` always be set here, or could this possibly result in a key error?

Reply:
> Also, will extra_data['token_id'] always be set here, or could this possibly result in a key error?

yeah, it is always present.